mqtt/mqtt-broker-core/src/auth/mod.rs

```diff
@@ -44,4 +49,29 @@ impl<T: Into<Identity>> From<T> for AuthId {
 }
 
 /// Non-anonymous client identity.
-pub type Identity = String;
+#[derive(Clone, Debug, Eq, Hash, PartialEq)]
+pub struct Identity(Arc<String>);
```

Comment:
just curious: why this change?

Reply:
We are cloning `AuthId` a bunch of times during the client life time, so it should be don earlier.